inspirehep/modules/workflows/tasks/upload.py

```diff
@@ -49,15 +50,32 @@ def store_record(obj, *args, **kwargs):
     # holdingpen stack.
     record = InspireRecord.create(obj.data, id_=None)
 
+    # Workflow save `_files` with workflow objects files information
+    # When setting the record.files the `_files` are not overridden
+    # because in `FilesIterator.flush` method ignores the `_files`
+    # update if exists one in the record.
+
+    for key in obj.files.keys:
+        with open(os.path.realpath(obj.files[key].file.uri)) as stream:
+            record.files[key] = stream
+            key_updated = False
+            for document in record.get('documents', []):
```

Comment:
I added because I want to omit iterating to figures if I have already find the key. Probably there is a better way finding it.